front_end/webserver.py

```diff
@@ -203,14 +218,17 @@ def post(self, user_id):
         try:
             course_id = self.get_body_argument("course_id")
             passcode = self.get_body_argument("passcode")
+            consent_given = self.get_body_argument("consent_given") == "Yes"
 
             course_basics = content.get_course_basics(course_id)
             course_details = content.get_course_details(course_id)
             course_title = course_basics["title"]
             course_passcode = course_details["passcode"]
             result = ""
 
-            if (course_passcode == None or course_passcode == passcode):
+            if not consent_given:
+                result = "Error: Cannot register for this course without consent."
```

Comment:
This is a minor quibble, but let's not use "Error" in the message. In my mind, that would indicate that there is a bug in the system. But we're just wanting to let them know that they've done something that is not allowed. Instead, please change the wording to "You may not register for this course without providing consent to participate in the research study. Please review the instructions on what you should do if you decline to participate in the study."